packages/mongodb-chatbot-server/src/processors/generateResponseWithSearchTool.ts

```diff
@@ -0,0 +1,369 @@
+import {
+  References,
+  SomeMessage,
+  SystemMessage,
+  UserMessage,
+  AssistantMessage,
+  ToolMessage,
+  EmbeddedContent,
+} from "mongodb-rag-core";
+import { z } from "zod";
+import { GenerateResponse } from "./GenerateResponse";
+import {
+  CoreAssistantMessage,
+  CoreMessage,
+  LanguageModel,
+  streamText,
+  Tool,
+  ToolCallPart,
+  ToolChoice,
+  ToolExecutionOptions,
+  ToolResultUnion,
+  ToolSet,
+  CoreToolMessage,
+} from "mongodb-rag-core/aiSdk";
+import { FilterPreviousMessages } from "./FilterPreviousMessages";
+import { InputGuardrail, withAbortControllerGuardrail } from "./InputGuardrail";
+import { strict as assert } from "assert";
+import { MakeReferenceLinksFunc } from "./MakeReferenceLinksFunc";
+import { makeDefaultReferenceLinks } from "./makeDefaultReferenceLinks";
+import { SearchResult } from "./SearchResult";
+
+export const SEARCH_TOOL_NAME = "search_content";
+
+export type SearchToolReturnValue = {
+  content: SearchResult[];
+};
+
+export type SearchTool<ARGUMENTS extends z.ZodTypeAny> = Tool<
```

Comment:
nit - particular reason for the all caps `ARGUMENTS` type instead of `Arguments`?